test/e2e/tests/reload.go

```diff
@@ -0,0 +1,279 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build e2e
+// +build e2e
+
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, ReloadTest)
+}
+
+var ReloadTest = suite.ConformanceTest{
+	ShortName:   "Reload",
+	Description: "Envoy gateway reload route order",
+	Manifests:   []string{"testdata/reload-route-order.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		t.Run("Envoy gateway reload", func(t *testing.T) {
+			// Step 1: Start with an initial configuration for the Envoy Proxy
+
+			var namespace = "envoy-gateway-system"
+
+			initialConfig, err := getConfigDump(t, suite.RestConfig, suite.Client, namespace)
+			if err != nil {
+				t.Log(err)
+				t.Fatal(err) // Fail the test and log the error
+			}
+			var numReloads = 5
+			var numCheckingEachReload = 3
+			// Channel to communicate between Goroutines
+			errCh := make(chan error)
+
+			for i := 0; i < numReloads; i++ {
+				// Step 2: Restart or reload the Envoy Gateway in a Goroutine
+				go func() {
+					err := restartEnvoyGateway(t, suite.Client, namespace)
+					if err != nil {
+						errCh <- err
+						return
+					}
+					errCh <- nil // Notify that Step 2 is completed
+				}()
+
+				// Nested loop to check config dump multiple times during each reload
+				for j := 0; j < numCheckingEachReload; j++ {
+
+					// Step 3: Retrieve the `/config_dump` output from the Envoy Proxy admin interface
+					newConfigDump, err := getConfigDump(t, suite.RestConfig, suite.Client, namespace)
+					if err != nil {
+						t.Log(err)
+						t.Fatal(err)
+					}
+
+					// Step 4: Compare the obtained `/config_dump` output with the initial configuration
+					// Define options for comparison
+					sortOpts := cmpopts.SortSlices(func(a, b interface{}) bool {
```

Comment:
why are we not comparing everything and why are we sorting ?

Reply:
It fails if we compare without sorting.
Cluster 'accesslog' and cluster 'tracing' are the two values in 'synamic_endpoint_configs' array. The contents are same, but the order is different.

Please find more detailed log at https://github.com/envoyproxy/gateway/actions/runs/5638125714/job/15272037648#step:6:4166
![image](https://github.com/envoyproxy/gateway/assets/76408835/971b871c-1654-4abb-a04c-d335ecc2b358)

Reply:
this might be because we are marshaling into a `map`, is there a need to do that, can't we just compare the o/p of `/config_dump` as a string ?

Reply:
Sure I will double check on that and run one more test.

Reply:
@arkodg 
Please refer to the log from another draft PR, the test failed when comparing either bytes or string.
pr: https://github.com/envoyproxy/gateway/pull/1722/files#diff-7ceb4e766433e95b4b40c8b6cd6996a2c2146242d471d0eddee208b366b74aa3R82
log: https://github.com/envoyproxy/gateway/actions/runs/5685198922/job/15409803355?pr=1722#step:6:2685

Reply:
@arkodg 
I would like to follow up on this comment if you don't mind. Please let me know if you have any other questions.

Thanks.

Reply:
@arkodg 
I would like to follow up once again. Please let me know if there is anything else I can do for this PR.

Thanks.

Reply:
hi @Ronnie-personal, we need to investigate to see why these values are changing

Reply:
> hi @Ronnie-personal, we need to investigate to see why these values are changing

Thanks for the comment, I will think about it.